### PR TITLE
Add RandomCrop and CombinedImageTransform.

### PR DESCRIPTION
**Pull request by crockpotveggies**
Allows for random cropping of images and composing image pipelines of random variability.

Addresses https://github.com/deeplearning4j/DataVec/issues/217

Please merge when tests are added.

### CODE REVIEW

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ *  *
+ *  *    Unless required by applicable law or agreed to in writing, software
+ *  *    distributed under the License is distributed on an "AS IS" BASIS,
+ *  *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  *    See the License for the specific language governing permissions and
+ *  *    limitations under the License.
+ */
+package org.datavec.image.transform;
+
+import javafx.util.Pair;
```
Comment (raver119) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java]: No. No javafx please.

Use our Pair.

File: datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ *
+ * @author Justin Long (@crockpotveggies)
+ */
+public class CombinedImageTransform extends BaseImageTransform<opencv_core.Mat> {
+
+    protected List<Pair<ImageTransform, Double>> transforms;
+    protected boolean shuffle;
+
+    public CombinedImageTransform(List<Pair<ImageTransform, Double>> transforms) {
+        this(new Random(1234), transforms, false);
```
Comment (raver119) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java]: No, not a java Random please.

Use native Nd4j.getRandom() method, and use it instead.

Comment (crockpotveggies) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java]: there's a problem here. base class for ImageTransform requires a java random. I can still set this and choose not to use it

Comment (raver119) [inline datavec-data/datavec-data-image/src/main/java/org/datavec/image/transform/CombinedImageTransform.java]: Let's update base class then.

Nd4j.getRandom() follows seed/sequence defined within model initialization, so it's kinda nice option to have for reproducibility.

**Comment by crockpotveggies** [comment]
Updated as per Gitter chat. Tests passing.

**Comment by raver119** [comment]
I'll pick it up from here, and add some more methods.

**Comment by saudet** [comment]
Could we also add MultiImageTransform as a specialization of PipelineImageTransform just so we don't break the API?

It sounds like we could also have enhanced CropImageTransform instead of creating another RandomCropTransform class?

**Comment by raver119** [comment]
Damn, yea. I'll do it now.

**Comment by raver119** [comment]
Sorry, it's back now, backed by PupelineImageTransform.

**Pull request merged**